hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/ReadFileStreamPTest.java

```diff
@@ -60,65 +62,149 @@ public void setup() throws IOException {
     }
 
     @Test
-    public void when_watchType_new() throws IOException, InterruptedException {
+    public void when_watchType_new_then_ignorePreexisting() throws IOException, InterruptedException, ExecutionException {
         DAG dag = buildDag(WatchType.NEW);
 
+        // this is a pre-existing file, should not be picked up
         File file = createNewFile();
+        appendToFile(file,   "hello", "pre-existing");
+        sleepAtLeastMillis(50);
 
         Future<Void> jobFuture = instance.newJob(dag).execute();
+        // wait for the processor to initialize and pickup what it wants to
+        sleepAtLeastSeconds(2);
+
+        // check that nothing is picked up
+        assertEquals(0, list.size());
+
+        finishDirectory(jobFuture, file);
+    }
 
+    @Test
+    @Ignore //this does not work reliably due to ENTRY_CREATE might be reported before contents is written
+    public void when_watchType_new_then_pickupNew() throws IOException, InterruptedException, ExecutionException {
+        DAG dag = buildDag(WatchType.NEW);
+
+        Future<Void> jobFuture = instance.newJob(dag).execute();
+        // wait for the processor to initialize
         sleepAtLeastSeconds(2);
-        appendToFile(file, "hello", "world");
-        assertTrueEventually(() -> assertEquals(2, list.size()));
 
-        assertTrue(file.delete());
-        assertTrue(directory.delete());
-        assertTrueEventually(() -> assertTrue("job should be completed eventually", jobFuture.isDone()));
+        File file = new File(directory, randomName());
+        appendToFile(file, "hello");
+
+        assertTrueEventually(() -> assertEquals(1, list.size()), 2);
+
+        finishDirectory(jobFuture, file);
+    }
+
+    @Test
+    @Ignore //this does not work reliably due to ENTRY_CREATE might be reported before contents is written
```

Comment:
NEW really isn't. However, maybe, Linux only reports NEW after the file is closed by the writing process, I don't know. Maybe you can test...

REPROCESS is the most correct, if you are ready to remove all data generated by the previous run. However, when line is emitted, you don't know, which file it is from (such as the `docId` needed in `TfIdf`).

APPEND is quite useful, with single limitation, that it can break words.

Anyway, the class is just for quick and dirty tests. Including the fact, that during heavy load the events might OVERFLOW.